numpy/core/shape_base.py

```diff
@@ -179,6 +171,100 @@ def atleast_3d(*arys):
         return res
 
 
+def atleast_nd(ary, ndim, pos=0, inplace=False):
+    """
+    View input as array with at least `ndim` dimensions.
+
+    New unit dimensions are inserted at the index given by `pos` if
+    necessary.
+
+    Parameters
+    ----------
+    ary : array_like
+        The input array. Non-array inputs are converted to arrays.
+        Arrays that already have `ndim` or more dimensions are
+        preserved.
+    ndim : scalar
+        The minimum number of dimensions required.
+    pos : int, optional
+        The index to insert the new dimensions. May range from
+        ``-ary.ndim`` to ``+ary.ndim``. Non-negative indices indicate
```

Comment:
Not true - the lower bound is `-ary.ndim - 1`

Reply:
Thanks for the catch. I had a feeling at the back of my head that the indices should not be the same, but was in too much of a hurry to place it. Fixed now.